storage/entities/transaction.js

```diff
@@ -653,6 +664,24 @@ class Transaction extends BaseEntity {
 
 		return filters;
 	}
+
+	/**
+	 * Checks if a dapp exists.
+	 *
+	 * @param {Object} params
+	 * @param {string} params.transactionId
+	 * @param {string} params.name
+	 * @param {string} params.link
+	 * @returns {Promise}
+	 * @todo This method should be replaced with calls to storage get() and correct filters
+	 */
+	getExistingDapps(filters = {}, tx) {
+		const sql = `SELECT name, link FROM dapps
+		WHERE (name = '${filters.name}' OR link = '${
+			filters.link
+		}') AND "transactionId" != '${filters.transactionId_ne}'`;
+		return this.adapter.execute(sql, {}, {}, tx);
```

Comment:
This will cause SQL injection because you are not sanitizing the parameters and directly concatenating in the query. The right way to do this is: 

```
const sql = 'SELECT name, link, FROM dapps WHERE name = ${name} OR link = ${link} AND transactionId != ${transactionId}'
return this.adapter.execute(sql, filters, {}, tx);
```

Reply:
Removed method and logic is now using `get()`